core/domain/exp_domain.py

```diff
@@ -276,8 +278,13 @@ class ExplorationChange(change_domain.BaseChange):
     }, {
         'name': CMD_ADD_TRANSLATION,
         'required_attribute_names': [
-            'state_name', 'content_id', 'language_code', 'content_html',
-            'translation_html'],
+            'state_name', 'content_id', 'language_code',
+            'needs_update', 'translation_html', 'data_format'],
+        'optional_attribute_names': ['content_html'],
+        'user_id_attribute_names': []
+    }, {
+        'name': CMD_MARK_TRANSLATION_AS_NEEDING_UPDATE,
+        'required_attribute_names': ['content_id', 'state_name'],
```

Comment:
Do we need to know the language code as well? [Assume a user marked a Hindi translation needs update, how are we going to handle that?]

Reply:
Can you please explain how to mark a particular language as needing update? I haven't tried this yet.

Reply:
Per offline discussion, we found that there is currently no way to mark a particular language as needing update via the UI. Also, there is no strong reason to support this in the future.